### ISSUE TITLE
Cannot install lantern dependencies on aarm64 architecture

### ISSUE THREAD

**Issue by bagdaddy** [mlverse/torch#1182]
Hello, whenever I try to install and use torch, it says that I don't have certain dependencies installed (lantern).
I went through all documentation and I still don't get how I can achieve that manually. 

When I try to get the physical files for manual installation, I get an error that my architecture is unsupported. I'm running a rocker/rstudio docker container on an M1 Mac computer.

`> library(torch)
ℹ Additional software needs to be downloaded and installed for torch to work correctly.
Warning message:
ℹ Failed to install torch, manually run `install_torch()`
✖ Currently only `x86_64` systems are supported for automatic installation.
Caused by error in `check_can_autoinstall()`:
! Currently only `x86_64` systems are supported for automatic installation.
ℹ You can manually compile LibTorch for you architecture following instructions in <8;;https://github.com/pytorch/pytorch#from-sourcehttps://github.com/pytorch/pytorch#from-source>
ℹ You can then use `install_torch_from_file()` to install manually. `

What I tried:

- installing using install.packages()
- installing straight from github
- installing manually, but I can't seem to get a lantern library URL to download it from, because the helper functions simply do not work on my machine.

When trying to run torch::install_torch() or  I get the following error: 
`! Architecture "aarch64" is not supported in this OS.`

I looked at the code, is this issue due to me running this in a docker container? 
My architecture is aarch64, but the "OS" in the container is ubuntu. How can I work around this?

Maybe someone could point me to how I can download the actual lantern lib or something? I will set the env variables for torch and lantern urls and maybe then it will get installed? I am starting to go crazy over this

**Comment by bagdaddy** [comment]
Thank you for such a rapid response!

Do you mean to point to the correct python installation? Or do I need to refer to the version of R torch package? The container is running 3.10 python version.

**Comment by dfalbel** [comment]
Sorry, I meant pointing to the correct LibTorch version (currently it's 2.0.1), when building LibTorch from source:

https://github.com/mlverse/torch/blob/fe5020f62e6aa6c3c7fbce06c4aacb1e09c55db7/src/lantern/CMakeLists.txt#L4

**Comment by bagdaddy** [comment]
Wouldn't the version be decided by the branch name I provide when cloning the pytorch repository? In this case it would be the latest ~2.4, should I set the branch to v2.0.1 instead?

**Comment by dfalbel** [comment]
Yes, you have to clone from the tag v2.0.1 in order to get the correct LibTorch version.

**Comment by bagdaddy** [comment]
Unfortunately I'm getting this error when building the sources:
```
CMake Error at cmake/Dependencies.cmake:926 (if):
  if given arguments:

    "VERSION_LESS" "3"
```

I tried using this to fix these errors, https://github.com/shanemcandrewai/pytorch-setup?tab=readme-ov-file, but it's not compatible with 2.0.1

Also, when I tried cloning from the main branch I didn't get these errors and I managed to run all of the steps successfully... It's when I tried to downgrade when it stopped working
